File: Backend/src/Sofco.Model/Models/Billing/Solfac.cs
using System;
using Sofco.Model.Enums;
using Sofco.Model.Utils;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using Sofco.Model.Models.Admin;

namespace Sofco.Model.Models.Billing
{
    public class Solfac : BaseEntity
    {
        public Solfac()
        {
            Hitos = new Collection<Hito>();
            Histories = new Collection<SolfacHistory>();
            StartDate = DateTime.Now;
        }
        
        public string BusinessName { get; set; }
        public string CelPhone { get; set; }
        public SolfacStatus Status { get; set; }
        public string ContractNumber { get; set; }

        public decimal Amount { get; set; }
        public decimal Iva21 { get; set; }
        public decimal TotalAmount { get; set; }

        public decimal CapitalPercentage { get; set; }
        public decimal BuenosAiresPercentage { get; set; }
        public decimal OtherProvince1Percentage { get; set; }
        public decimal OtherProvince2Percentage { get; set; }
        public decimal OtherProvince3Percentage { get; set; }
        public int Province1Id { get; set; }
        public int Province2Id { get; set; }
        public int Province3Id { get; set; }
        public string ParticularSteps { get; set; }
        public short TimeLimit { get; set; }
        public string Analytic { get; set; }

        public IList<Hito> Hitos { get; set; }
        public ICollection<SolfacHistory> Histories { get; set; }

        public int CurrencyId { get; set; }
        public Currency Currency { get; set; }

        public int? InvoiceId { get; set; }
        public Invoice Invoice { get; set; }

        public int DocumentTypeId { get; set; }
        public DocumentType DocumentType { get; set; }

        public int UserApplicantId { get; set; }
        public User UserApplicant { get; set; }

        public string ImputationNumber1 { get; set; }
        public ImputationNumber ImputationNumber { get; set; }
        public int ImputationNumber3Id { get; set; }

        public DateTime StartDate { get; set; }
        public int ModifiedByUserId { get; set; }
        public DateTime UpdatedDate { get; set; }

        public string ProjectId { get; set; }
        public string Project { get; set; }

        public string CustomerId { get; set; }
        public string ClientName { get; set; }

        public string ServiceId { get; set; }
        public string Service { get; set; }

        public List<SolfacAttachment> Attachments { get; set; }
    }
}

File: Frontend/src/models/billing/solfac/solfac.ts
import { HitoDetail } from 'models/billing/solfac/hitoDetail';

export class Solfac {
    constructor(
        public businessName: string,
        public clientName: string,
        public celphone: string,
        public statusName: string,
        public statusId: number,
        public contractNumber: string,
        public project: string,
        public documentTypeId: number,
        public userApplicantId: number,
        public imputationNumber1: number,
        public imputationNumber2: number,
        public imputationNumber3: number,
        public amount: number,
        public iva21: number,
        public currencyId: number,
        public totalAmount: number,
        public capitalPercentage: number,
        public buenosAiresPercentage: number,
        public otherProvince1Percentage: number,
        public province1Id: number,
        public otherProvince2Percentage: number,
        public province2Id: number,
        public otherProvince3ercentage: number,
        public province3Id: number,
        public hitos: HitoDetail[],
        public attachedParts: string,
        public particularSteps: string,
        public timeLimit: string
    ){}
}